src/core/pick.ts

```diff
@@ -1,24 +1,134 @@
 import type { StateTree } from 'pinia'
+import type { Mergeable } from '~/core/types'
+
+function merge(destination: Mergeable, source: Mergeable): Mergeable {
+  const mergingArrays = Array.isArray(destination) && Array.isArray(source)
+  const mergingObjects =
+    typeof destination === 'object' &&
+    typeof source === 'object' &&
+    destination !== null &&
+    source !== null
+
+  if (!mergingArrays && !mergingObjects) {
+    throw new Error('Can only merge object with object or array with array')
+  }
+
+  const result = (mergingArrays ? [] : {}) as Mergeable
+
+  const keys: string[] = [...Object.keys(destination), ...Object.keys(source)]
+  keys.forEach((key: string) => {
+    if (Array.isArray(destination[key]) && Array.isArray(source[key])) {
+      return (result[key] = [
```

Comment:
maybe avoid returning when the function is a `=> void` function and use if/else if instead ? 
not a fan of returning assignments

Reply:
Yes, very good point. Thanks for bringing this to my attention. Will improve this for sure. 🙂💪